R/color_palette.R

```diff
@@ -0,0 +1,107 @@
+#' Color Palettes Used in NEST
+#'
+#' @description `r lifecycle::badge("stable")`
+#'
+#' A standardized color palette to be used for all plots within
+#' the NEST project.
+#'
+#' @param n (\code{numeric} value) \cr
+#'   The number of colors to be returned from
+#'   the color palettes. Please note the colors will be repeated after
+#'   a certain numerical limit per palette:
+#'
+#'   \itemize{
+#'      \item{\code{nest}} {96 colors}
+#'      \item{\code{stream}} {38 colors}
+#'      \item{\code{viridis}} {49 colors}
+#'   }
+#'
+#' @param palette (\code{character} value)\cr
+#'   The name of a palette supported by this function
+#'
+#'   \itemize{
+#'      \item{\code{nest}} {A color palette developed by the NEST team representing
+#'       colors in the NEST logo.
+#'     }
+#'     \item{\code{stream}} {A color palette proposed by the stream documentation of stream version \code{2_05}}
+#'        \item{\code{viridis}} {A color palette provided by the \code{viridis::plasma} function.}
+#'   }
+#'
+#' @examples
+#' library(grid)
+#'
+#' plot_pal <- function(x) {
+#'   grid.newpage()
+#'   pushViewport(plotViewport(rep(1, 4)))
+#'   pushViewport(viewport(layout = grid.layout(1, ncol = length(x))))
+#'   for (i in seq_along(x)) {
+#'     grid.rect(
+#'       gp = gpar(fill = x[i], col = NA),
+#'       vp = viewport(layout.pos.col = i, layout.pos.row = 1)
+#'     )
+#'   }
+#' }
+#'
+#' plot_pal(color_palette(n = 10, palette = "nest"))
+#'
+#' plot_pal(color_palette(n = 10, palette = "stream"))
+#'
+#' plot_pal(color_palette(n = 10, palette = "viridis"))
+#'
+#' @export
+color_palette <- function(n = 10, palette = "nest") {
+  checkmate::assert_string(palette)
+  match.arg(palette, c("nest", "stream", "viridis"))
+  checkmate::assert_number(n)
+
+  set.seed(124)
+  colors <- if (palette == "viridis") {
+    rep(c(
+      scales::col_bin(
+        c(
+          "#0D0887FF", "#1C068DFF", "#270592FF", "#310597FF", "#3B049AFF", "#44039EFF", "#4C02A1FF",
+          "#5502A4FF", "#5D01A6FF", "#6600A7FF", "#6E00A8FF", "#7601A8FF", "#7E03A8FF", "#8606A6FF",
+          "#8E0BA5FF", "#9511A1FF", "#9C179EFF", "#A21D9AFF", "#A92395FF", "#B02991FF", "#B52F8CFF",
+          "#BC3488FF", "#C13B82FF", "#C6417DFF", "#CC4678FF", "#D04D73FF", "#D5536FFF", "#DA596AFF",
+          "#DE5F65FF", "#E26561FF", "#E56B5DFF", "#E97257FF", "#ED7953FF", "#F07F4FFF", "#F3864AFF",
+          "#F68D45FF", "#F89441FF", "#FA9B3CFF", "#FCA338FF", "#FDAB33FF", "#FDB32FFF", "#FEBB2BFF",
+          "#FDC328FF", "#FDCC26FF", "#FBD424FF", "#F9DD25FF", "#F6E726FF", "#F3EF27FF", "#F0F921FF"
+        ),
+        1:49
+      )(sample(1:49, 49)),
+      ceiling(n / 49)
+    ))
```

Comment:
What's the point of randomization if it will always return the same thing? Might as well just put in a fixed order/sample, right?

Reply:
This has been removed and replaced with a fixed order vector